### ISSUE TITLE
A question about the chinese regex

### ISSUE THREAD

**Issue by godt21** [wazuh/wazuh#10018]
|4.15| Component | Action type |
|---| --- | --- |
 | Decoders | Error |

## Description
Hi, teams , I recently make a custome decode, and I found the wazuh seems like not support chinese regex

### Log examples
`日志类型:系统操作, 用户:admin(local), IP地址:10.10.10.10, 操作对象:用户登录, 操作类型:登录, 描述:admin(local)  10.10.10.10  登录成功	`							

### Docode xml
```
<decoder name="sangfor_waf_1">
    <prematch>日志类型</prematch>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex>日志类型:(\S+)</regex>
    <order>logtype</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex>用户:(\S+),</regex>
    <order>username</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex>IP地址:(\S+),</regex>
    <order>src_ip</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex>操作对象:(\S+),</regex>
    <order>operation_target</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex>操作类型:(\S+),</regex>
    <order>operation_type</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex>IP地址:(\S+),</regex>
    <order>描述:(\.*)</order>
</decoder>
```
Regards

**Comment by jcruzlp** [comment]
Hello @godt21

Yes, Wazuh does support the Chinese characters in regex, but in order to do that a few modifications must be done to your current decoder, which to be honest was very close to hitting the spot, so good work with it!

the `<regex>` field must be set to pcre2, which is the kind of regex that supports the Chinese characters, leaving it like `<regex type="pcre2">.

Also, you'll need to add `(*UTF)` before the regex to indicate that is in UTF-8 format, so leaving your decoder like the following (by @jnasselle) :
```xml
<decoder name="sangfor_waf_1">
    <prematch type="pcre2">(*UTF)日志类型</prematch>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex type="pcre2">(*UTF)日志类型:(\S+)</regex>
    <order>logtype</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex type="pcre2">(*UTF)用户:(\S+),</regex>
    <order>username</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex type="pcre2">(*UTF)IP地址:(\S+),</regex>
    <order>src_ip</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex type="pcre2">(*UTF)操作对象:(\S+),</regex>
    <order>operation_target</order>
</decoder>
<decoder name="sangfor_waf_1_child">
    <parent>sangfor_waf_1</parent>
    <regex type="pcre2">(*UTF)>操作类型:(\S+),</regex>
    <order>operation_type</order>
</decoder>
```
And after testing it with `/var/ossec/bin/wazuh-logtest` and your sample log I got this result:
```bash
root@ubuntu20LTS:/home/vagrant# /var/ossec/bin/wazuh-logtest
Starting wazuh-logtest v4.2.1
Type one log per line

日志类型:系统操作, 用户:admin(local), IP地址:10.10.10.10, 操作对象:用户登录, 操作类型:登录, 描述:admin(local) 10.10.10.10 登录成功 

**Phase 1: Completed pre-decoding.
	full event: '日志类型:系统操作, 用户:admin(local), IP地址:10.10.10.10, 操作对象:用户登录, 操作类型:登录, 描述:admin(local) 10.10.10.10 登录成功 '

**Phase 2: Completed decoding.
	name: 'sangfor_waf_1'
	logtype: '系统操作,'
	operation_target: '用户登录'
	src_ip: '10.10.10.10'
	username: 'admin(local)'
```

I hope this could help you, if you have any further questions or doubts, please do not hesitate to ask!
Best regards.

**Comment by jcruzlp** [comment]
Also worth noting, there may be weak links between components, so if you could check that it works on the Wazuh Web Interface would be great.

**Comment by godt21** [comment]
Thanks for your answer, it is useful help for me , and I met another question, my log is like a json, and  I want to use my own decode, but the wazuh still use the json to decode my log, how do I imporove it?
Regards.

**Comment by jcruzlp** [comment]
Hello again! 
To use a more dedicated decoder instead of the JSON, you need to use a decoder like the one you used in your original question, this will extract whatever you indicate in it by regex and in the field under the `<order>` field.

**Comment by godt21** [comment]
> Hello again!
> To use a more dedicated decoder instead of the JSON, you need to use a decoder like the one you used in your original question, this will extract whatever you indicate in it by regex and in the field under the `<order>` field.
I don't understand your answer about use a decoder like the one you used in your original question can you make a easy example?
regards.

**Comment by godt21** [comment]
I am on the basis of the wazuh document to remove json decode, so I used the  `<plugin_decoder offset="after_prematch">JSON_Decoder</plugin_decoder>`, and I use the original  way to decode it told the same error,such as
the decode:

```
<decoder name="sangfor_waf_11">
    <prematch>(*UTF)日志</prematch>
</decoder>
<decoder name="sangfor_waf_11_child">
    <parent>sangfor_waf_11</parent>
    <regex type="pcre2">(*UTF)日志类型:(\S+),</regex>
    <order>logtype</order>
</decoder>
```

the result:
```
2021/08/29 00:34:49 ossec-testrule: INFO: Started (pid: 570).

Since Wazuh v4.1.0 this binary is deprecated. Use wazuh-logtest instead

ossec-testrule: Type one log per line.

{"message":"日志类型:用户认证, 用户:libl1, IP地址:10.101.194.219, 操作对象:注销, 登录时间:2019-07-31 10:22:53, 登录时长:8, 注销时间:10:23:01\n","@version":"1","@timestamp":"2019-07-31T02:23:01.000Z","type":"syslog","host":"10.101.227.30","priority":134,"timestamp":"Jul 31 10:23:01","logsource":"localhost","program":"fwlog","severity":6,"facility":16,"facility_label":"local0","severity_label":"Informational"}					


**Phase 1: Completed pre-decoding.
       full event: '{"message":"日志类型:用户认证, 用户:libl1, IP地址:10.101.194.219, 操作对象:注销, 登录时间:2019-07-31 10:22:53, 登录时长:8, 注销时间:10:23:01\n","@version":"1","@timestamp":"2019-07-31T02:23:01.000Z","type":"syslog","host":"10.101.227.30","priority":134,"timestamp":"Jul 31 10:23:01","logsource":"localhost","program":"fwlog","severity":6,"facility":16,"facility_label":"local0","severity_label":"Informational"}	   '
       timestamp: '(null)'
       hostname: 'manager'
       program_name: '(null)'
       log: '{"message":"日志类型:用户认证, 用户:libl1, IP地址:10.101.194.219, 操作对象:注销, 登录时间:2019-07-31 10:22:53, 登录时长:8, 注销时间:10:23:01\n","@version":"1","@timestamp":"2019-07-31T02:23:01.000Z","type":"syslog","host":"10.101.227.30","priority":134,"timestamp":"Jul 31 10:23:01","logsource":"localhost","program":"fwlog","severity":6,"facility":16,"facility_label":"local0","severity_label":"Informational"}		   '

**Phase 2: Completed decoding.
       decoder: 'json'
       message: '日志类型:用户认证, 用户:libl1, IP地址:10.101.194.219, 操作对象:注销, 登录时间:2019-07-31 10:22:53, 登录时长:8, 注销时间:10:23:01
'
       @version: '1'
       @timestamp: '2019-07-31T02:23:01.000Z'
       type: 'syslog'
       host: '10.101.227.30'
       priority: '134'
       timestamp: 'Jul 31 10:23:01'
       logsource: 'localhost'
       program: 'fwlog'
       severity: '6'
       facility: '16'
       facility_label: 'local0'
       severity_label: 'Informational'
```
It still decode in json,but I do not want the json to decode, the right decode should be ` sangfor_waf_11`
